src/components/select/filterable-select/filterable-select.spec.js

```diff
@@ -947,6 +947,35 @@ describe("FilterableSelect", () => {
         .simulate("change", { target: { value: filterText } });
       expect(onFilterChangeFn).toHaveBeenCalledWith(filterText);
     });
+
+    it("then it should not be invoked when the component rerenders", () => {
+      const Component = () => {
+        const [onFilterChangeCalled, setOnFitlerChangeCalled] = useState(false);
+
+        return (
+          <>
+            <div id="on-filter-called">
+              {onFilterChangeCalled ? "true" : "false"}
+            </div>
+            <FilterableSelect
+              name="testSelect"
+              id="testSelect"
+              onFilterChange={() => setOnFitlerChangeCalled(true)}
+            >
+              <Option value="opt1" text="red" />
+              <Option value="opt2" text="green" />
+              <Option value="opt3" text="blue" />
+              <Option value="opt4" text="black" />
+            </FilterableSelect>
+          </>
+        );
+      };
+
+      const wrapper = mount(<Component />);
+      expect(wrapper.find("#on-filter-called").text()).toBe("false");
```

Comment:
not criticising the approach, but I'm curious why you've made it so that calling `onFilterChange` updates the component text and testing for that, rather than using a mock and checking that's not been called, which is our usual approach? They're both testing for the same thing, just curious why you've adopted a different approach here from what we do elsewhere.

Reply:
oh I see. Sorry I should have thought of that!